plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Adapter.java

```diff
@@ -34,28 +34,40 @@
     static class StatementMetadata {
         final ParserRuleContext source;
 
-        boolean last;
+        boolean lastSource;
+        boolean topLoop;
+        boolean inLoop;
+        boolean lastLoop;
 
-        boolean allExit;
-        boolean allReturn;
-        boolean anyReturn;
-        boolean allBreak;
-        boolean anyBreak;
-        boolean allContinue;
+        boolean methodEscape;
```

Comment:
It'd really help to have javadoc on these so I can figure out what they do while scanning the PR.